includes/utilities.inc

```diff
@@ -47,58 +47,396 @@ function islandora_openseadragon_construct_clip_url($params, $download = FALSE)
+    'zoomOutButton' => NULL,
+    'homeButton' => NULL,
+    'fullPageButton' => NULL,
+    'rotateLeftButton' => NULL,
+    'rotateRightButton' => NULL,
+    'previousButton' => NULL,
+    'nextButton' => NULL,
+    'sequenceMode' => FALSE,
+    'initialPage' => 0,
+    'preserveViewport' => FALSE,
+    'preserveOverlays' => FALSE,
+    'showReferenceStrip' => FALSE,
+    'referenceStripScroll' => 'horizontal',
+    'referenceStripElement' => NULL,
+    'referenceStripHeight' => NULL,
+    'referenceStripWidth' => NULL,
+    'referenceStripPosition' => 'BOTTOM_LEFT',
+    'referenceStripSizeRatio' => 0.2,
+    'collectionMode' => FALSE,
+    'collectionRows' => 3,
+    'collectionColumns' => 0,
+    'collectionLayout' => 'horizontal',
+    'collectionTileSize' => 800,
+    'collectionTileMargin' => 80,
+    'crossOriginPolicy' => FALSE,
+    'ajaxWithCredentials' => FALSE,
+  );
+  return $defaults;
+}
+
+/**
+ * Image metadata pertinent to the users chosen TileSources implementation.
+ *
+ * @param AbstractDatastream $datastream
+ *   The datastream for which to fetch the tile source implementation.
+ *
+ * @return array
+ *   The metadata required by the users chosen implementation.
+ */
+function islandora_openseadragon_tile_source(AbstractDatastream $datastream) {
+  switch (variable_get('islandora_opensedragon_tilesource', 'djatoka')) {
+    case 'djatoka':
+      return islandora_openseadragon_djatoka_tile_source($datastream);
+
+    case 'iiif':
+      return islandora_openseadragon_iiif_tile_source($datastream);
+
+    default;
+      return array();
+  }
+}
+
+/**
+ * Image metadata pertinent to the IIIF TileSources implementation.
+ *
+ * @param AbstractDatastream $datastream
+ *   Gets the metadata for viewing the given datastream, it assumes that the
+ *   datastream is an image and it's format is supported by the IIIF server.
+ *
+ * @return array
+ *   An associative array with the required metadata for AdoraDjatoka tile
+ *   source to be rendered.
+ *   - identifier: A URL where the datastream image can be accessed.
+ *   - width: The width of the image in pixels if known.
+ *   - height: The height of the image in pixels if known.
+ *   - tileWidth: The width of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileHeight: The height of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileOverlap: The number of pixels each tile is expected to overlap
+ *     touching tiles.
+ *   - minLevel: The minimum level to attempt to load.
+ *   - maxLevel: The maximum level to attempt to load.
+ */
+function islandora_openseadragon_iiif_tile_source(AbstractDatastream $datastream) {
+  $base_url = variable_get('islandora_openseadragon_iiif_url', 'loris');
+  $pid = $datastream->parent->id;
+  $dsid = $datastream->id;
+  // Currently Loris works straight out of fedora only.
+  return url("{$base_url}/{$pid}/datastreams/{$dsid}/", array(
+    'absolute' => TRUE,
```

Comment:
Does this have to be `absolute`? It doesn't make sense to call `url('loris', array('absolute' => TRUE));` as Drupal may assume it's responsible for its routing and do silly things with languages and/or multisites (should probably be `/loris` (or `//loris`), to indicate that it's relative to the host).

I could see doing the language hack, (something like: `'language' => language_default()`, should suffice?) to avoid potentially trying to inject language prefix info if the current language is different than the default, if there _was_ a Drupal menu path being use to serve IIIF.